### PR TITLE
Dashboard data and Vehicle selection

### PR DESCRIPTION
**Pull request by haydenf**
Once logged in or page refreshed it'll make you select a vehicle but won't display data cards and dashboard data now shows the relevant data depending on what vehicle has been selected (the only data that changes for now is trip data because it's the only one that has different data) Phillip and Haydens work from today.

### CODE REVIEW

File: client/src/Components/CarData.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import TimeInCarCard from './Card/TimeInCarCard';
+import EmissionsCard from './Card/EmissionsCard';
+import FuelEconomyCard from './Card/FuelEconomyCard';
+import GoogleMap from './GoogleMap/GoogleMap';
+import AntdTableComponent from './AntdTable/AntdTableComponent';
+
+export default function CarData() {
+  const { loading, error, data } = useQuery(getDashboardAllDataQuery, {
+    variables: {
+      vehicleId: `27e1dabc-a89a-444d-bcc3-ace5a33a3d26`,
```
Comment (jeznag) [inline client/src/Components/CarData.js]: can we remove this hardcoded value?

File: client/src/Components/CarData.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    },
+  });
+
+  if (loading) {
+    return <LoadingSpinner />;
+  } else if (error) {
+    debugger;
```
Comment (jeznag) [inline client/src/Components/CarData.js]: let's remove the debugger

File: client/src/Components/DashboardHeader.js [hunk 3]
```diff
@@ -1,30 +1,44 @@
...
+    return (
+      <div className="hero-img">
+        <Header as="h1">Welcome back!</Header>
+        <div className="hero-img-car-data">
+          <h3> Cars Life Statistics</h3>
+          <Divider hidden />
+          <Statistic.Group size="mini">
+            <Statistic>
+              <Statistic.Value>{data.car.trips.length}</Statistic.Value>
+              <Statistic.Label> Total trips </Statistic.Label>
```
Comment (jeznag) [inline client/src/Components/DashboardHeader.js:32]: Might want to remove the space before Total

File: client/src/Components/DashboardHeader.js [hunk 4]
```diff
@@ -1,30 +1,44 @@
...
+    },
+  });
+  if (loading) {
+    return <LoadingSpinner />;
+  } else if (error) {
+    return <p>Error! {error}</p>;
+  } else {
+    return (
+      <div className="hero-img">
+        <Header as="h1">Welcome back!</Header>
```
Comment (jeznag) [inline client/src/Components/DashboardHeader.js]: can we put the user's first name here?

File: client/src/Components/DashboardHeader.js [hunk 5]
```diff
@@ -1,30 +1,44 @@
...
+        <div className="hero-img-car-data">
+          <h3> Cars Life Statistics</h3>
```
Comment (jeznag) [inline client/src/Components/DashboardHeader.js]: Maybe `Car's Lifetime Statistics from GOFAR`

File: client/src/Pages/Dashboard/DashboardView.js [hunk 6]
```diff
@@ -1,87 +1,33 @@
...
-  const [vehicleIdSelectionState, setVehicleIdSelectionState] = useState(
-    '27e1dabc-a89a-444d-bcc3-ace5a33a3d26'
-  );
-  const handleChangeOnVehicleDropdown = vehicleIdSelectionState => {
-    setVehicleIdSelectionState(vehicleIdSelectionState);
+  const [vehicleIdSelectionState, setVehicleIdSelectionState] = useState(null);
+
+  const selectVehicle = vehicleIdSelectionState => {
+    setVehicleIdSelectionState(vehicleIdSelectionState.value);
+    console.log(vehicleIdSelectionState);
```
Comment (jeznag) [inline client/src/Pages/Dashboard/DashboardView.js]: can you remove this console.log

File: client/src/Pages/Dashboard/DashboardView.js [hunk 7]
```diff
@@ -1,87 +1,33 @@
...
-          <Card.Group>
-            <EmissionsCard emission={data.car.emissions} />
-            <FuelEconomyCard fuelEconomy={data.car.fuelEconomy} />
-          </Card.Group>
-        </div>
-        <div className="ui container">
-          <AntdTableComponent />
+        <CarList test={selectVehicle} />
+        <div style={{ height: '500px', textAlign: 'center' }}>
+          <h1>you must select a car</h1>
```
Comment (jeznag) [inline client/src/Pages/Dashboard/DashboardView.js]: Capital Y

File: client/src/Pages/Login/LoginView.js [hunk 8]
```diff
@@ -26,18 +29,19 @@ const LoginView = props => {
...
     props.handleLoggedIn({
       authToken: data.login.authToken,
-      userId: data.login.userId
+      userId: data.login.userId,
     });
   }
 
   return (
     <>
+      {console.log(data, 'this is the data from loginviewq')}
```
Comment (jeznag) [inline client/src/Pages/Login/LoginView.js]: remove. Can you make sure there is an eslint rule that prevents console.log? (`no-console` rule)

File: client/src/Components/DashboardHeader.js (hunk 4 above)
Comment (haydenf) [reply client/src/Components/DashboardHeader.js]: Will resolve this tomorrow, need a bit more time to do it

File: client/src/Components/CarData.js [hunk 9]
```diff
@@ -15,17 +15,16 @@ import FuelEconomyCard from './Card/FuelEconomyCard';
 import GoogleMap from './GoogleMap/GoogleMap';
 import AntdTableComponent from './AntdTable/AntdTableComponent';
 
-export default function CarData() {
+export default function CarData(props) {
   const { loading, error, data } = useQuery(getDashboardAllDataQuery, {
     variables: {
-      vehicleId: `27e1dabc-a89a-444d-bcc3-ace5a33a3d26`,
+      vehicleId: `${props.state}`,
```
Comment (jeznag) [inline client/src/Components/CarData.js]: state is not a good prop name in my opinion. I would suggest props.vehicleId

**Review by jeznag — APPROVED** [review]

File: client/src/Components/CarData.js (hunk 9 above)
Comment (haydenf) [reply client/src/Components/CarData.js]: changed to vehicleId

File: client/src/Components/CarData.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+import TimeInCarCard from './Card/TimeInCarCard';
+import EmissionsCard from './Card/EmissionsCard';
+import FuelEconomyCard from './Card/FuelEconomyCard';
+import GoogleMap from './GoogleMap/GoogleMap';
+import AntdTableComponent from './AntdTable/AntdTableComponent';
+
+export default function CarData(props) {
+  const { loading, error, data } = useQuery(getDashboardAllDataQuery, {
+    variables: {
+      vehicleId: props.vehicleIdState.value,
```
Comment (jeznag) [inline client/src/Components/CarData.js:21]: nitpick `props.vehicleId` would have been fine. It's confusing using the word state for something that is a prop not state.

**Review by jeznag — APPROVED** [review]

**Pull request merged**